db.go

```diff
@@ -161,6 +168,25 @@ func (db *Db) setActiveDataFile(activeDataFile string) error {
 	return nil
 }
 
+func (db *Db) getSegmentFilePointerFromPath(path string) (*os.File, error) {
```

Comment:
Wherever path is used, explicitly specify if its a active or inactive file. Then it'll just be one call worst case. Challenge is to handle if the active file becomes inactive before any operation. But that is handled since we use Locks.